synfig-studio/src/gui/canvasview.cpp

```diff
@@ -1543,6 +1534,35 @@ CanvasView::init_menus()
 
 }
 
+void
+CanvasView::init_plugins()
+{
+	auto instance = get_instance().get();
+	if(instance)
+		for ( const auto& plugin : App::plugin_manager.plugins() )
+		{
+			std::string id = plugin.id;
+			if(!action_group->get_action(id))
+				action_group->add(
+					Gtk::Action::create(id, plugin.name.get()),
+					[instance, id](){instance->run_plugin(id, true);}
+				);
+		}
+
+	// remove group if exists
+	typedef std::vector< Glib::RefPtr<Gtk::ActionGroup> > ActionGroupList;
+	ActionGroupList groups = App::ui_manager()->get_action_groups();
+	for(ActionGroupList::const_iterator it = groups.begin(); it != groups.end(); ++it)
```

Comment:
As you  don't need the iterator `it` itself, you can use range-based `for` here too:

```c++
for (const auto& group : groups) {
```